GliaWidgets/Glia.swift

```diff
@@ -1,35 +1,147 @@
+///     2. Add following privacy descriptions to your app's Info.plist:
+///         Privacy - Microphone Usage Description
+///         Privacy - Camera Usage Description
+///         Privacy - Photo Library Additions Usage Description
+///     3. Import `GliaWidgets` and `SalemoveSDK`:
+///         import GliaWidgets
+///         import SalemoveSDK
+///     4. Create configuration:
+///         let conf = Configuration(
+///             appToken: "appToken",
+///             apiToken: "apiToken",
+///             environment: .europe,
+///             site: "site"
+///         )
+///     5. Create visitor context:
+///         let visitorContext = VisitorContext(
+///             type: .page,
+///             url: "https://www.yoursite.com"
+///         )
+///     6. Start the engagement:
+///         do {
+///             try Glia.sharedInstance.start(
+///                 .chat,
+///                 configuration: conf,
+///                 queueID: "queueID",
+///                 visitorContext: visitorContext
+///             )
+///         } catch {
+///             // configuration error
+///         }
+///
+/// To listen Glia's events, use `onEvent`:
+///
+///     Glia.sharedInstance.onEvent = { event in
+///         switch event {
+///         case .started:
+///             break
+///         case .engagementChanged(let kind):
+///             break
+///         case .ended:
+///             break
+///         case .minimized:
+///             break
+///         case .maximized:
+///             break
+///         }
+///     }
+///
+/// To use custom colors and fonts, create a new theme with `ThemeColor` and `ThemeFont`:
+///
+///     let color = ThemeColor(
+///         primary: primaryColorCell.color,
+///         ...
+///     )
+///     let font = ThemeFont(
+///         header1: header1FontCell.selectedFont,
+///         ...
+///     )
+///
+///     let colorStyle: ThemeColorStyle = .custom(color)
+///     let fontStyle: ThemeFontStyle = .custom(font)
+///
+///     let theme = Theme(
+///         colorStyle: colorStyle,
+///         fontStyle: fontStyle
+///     )
+///
+/// and pass it to the `start(...)` method.
+///
+/// To customize a theme, you can override its properties. For example:
+///
+///     let theme = Theme()
+///     theme.chat.backButton = HeaderButtonStyle(image: UIImage(), color: .red)
+///     theme.chat.messageEntry.messageColor = .blue
+///
 public class Glia {
+    /// A singleton to access the Glia's interface.
     public static let sharedInstance = Glia()
+
+    /// Current engagement kind.
     public var engagement: EngagementKind { return rootCoordinator?.engagementKind ?? .none }
+
+    /// Used to monitor engagement state changes.
     public var onEvent: ((GliaEvent) -> Void)?
 
     private var rootCoordinator: RootCoordinator?
 
     private init() {}
 
+    /// Starts the engagement.
+    ///
+    /// - Parameters:
+    ///   - engagementKind: Kind of the engagement.
```

Comment:
Engagement media type.

Reply:
Again, changed only the description itself for now. If it's not enough, will go through all of the files and change the name.